### DISCUSSION TITLE
Cant extract Pnp Provisioning template from Project Management site template

### DISCUSSION THREAD

**Discussion by nav67** [pnp/powershell#2421] [Q&A]
We are trying to extract the provisioning template from a site with Project Management template applied. We are using Connect-PnPOnline to connect to a site  I am able to connect to the site but when trying to export the provisioning template using: Get-PnpProvisioningTemplate -out "template.xml" we are receiving the following error. This does not occur on sites with no project management template applied 

Script failing when trying to run command with below error Get-PnPSiteTemplate -Out "PnP-Provisioning-File.xml" PS C:\Windows\system32] Get-PnPSiteTemplate -Out "PnP-Provisioning-File.xml" Get-PnPSiteTemplate : The JSON value could not be converted to System.Int32. Path: $.position.sectionIndex | LineNumber: 0 | BytePositionInLine: 107. At line:1 char:1 + Get-PnPSiteTemplate -Out "PnP-Provisioning-File.xml"

**Comment by AlyaKoni** [comment]
Same issue in C# .Net 6 with PnP.Framework 1.10.0: The JSON value could not be converted to System.Int32. Path: $.position.sectionIndex | LineNumber: 0 | BytePositionInLine: 107.

`            ProvisioningTemplateCreationInformation ptci
                        = new ProvisioningTemplateCreationInformation(srcContext.Web);
            ptci.IncludeAllClientSidePages = true;
            ptci.IncludeSearchConfiguration = false;
            ptci.IncludeAllTermGroups = false;
            ptci.IncludeContentTypesFromSyndication = false;
            ptci.IncludeNativePublishingFiles = false;
            ptci.IncludeSiteCollectionTermGroup = false;
            ptci.IncludeHiddenLists = false;
            ptci.IncludeSiteGroups = false;
            ptci.IncludeTermGroupsSecurity = false;
            ptci.ProgressDelegate = delegate (String message, Int32 progress, Int32 total)
            {
                logger.LogDebug("{0:00}/{1:00} - {2}", progress, total, message);
            };
            ProvisioningTemplate template = srcContext.Web.GetProvisioningTemplate(ptci);`

**Reply by AlyaKoni** [reply]
It's working, if I use only following handlers:
`
ptci.HandlersToProcess = Handlers.None;
ptci.HandlersToProcess ^= Handlers.Lists;
ptci.HandlersToProcess ^= Handlers.Fields;
ptci.HandlersToProcess ^= Handlers.ContentTypes;
`

**Reply by AlyaKoni** [reply]
The problematic handler which generates this json error is: PageContents

**Reply by nav67** [reply]
you mean this one ptci.IncludeAllClientSidePages = true? If you exclude Pages rest of content extracts fine?

**Reply by AlyaKoni** [reply]
This should work:
Get-PnPSiteTemplate -ExcludeHandlers PageContents

**Comment by haavardmeling** [comment]
I had the same problem. In my case it was related to a template I had created under "Site Pages\Templates". It wasn't something I needed so deleted the file and now it works. Didn't investigate any further. Could probably turn on some PnPTracelog to find the exact file causing the problem.

**Comment by stokuri** [comment]
I had the same problem. The issue is in [pnpcore](https://github.com/pnp/pnpcore) while Deserializing the [controlDataJson](https://github.com/pnp/pnpcore/blob/dev/src/sdk/PnP.Core/Model/SharePoint/Pages/Internal/CanvasControl.cs#L331).

The sample json for which the failure occurs in my case:

`controlDataJson: {"controlType":4,"id":"00000000-0000-0000-0000-100000000005","position":{"zoneIndex":1.0,"sectionIndex":2.0,"controlIndex":2.0,"sectionFactor":6,"layoutIndex":1},"emphasis":{}}`

`controlDataJson: {"controlType":3,"id":"00000000-0000-0000-0000-100000000006","position":{"zoneIndex":1.0,"sectionIndex":2.0,"controlIndex":3.0,"sectionFactor":6,"layoutIndex":1},"webPartId":"8654b779-4886-46d4-8ffb-b5ed960ee986","emphasis":{},"addedFromPersistedData":true}`

The parser is failing to read `"sectionIndex":2.0` property which is defend as `int` in [CanvasPosition.cs](https://github.com/pnp/pnpcore/blob/dev/src/sdk/PnP.Core/Model/SharePoint/Pages/Internal/CanvasPosition.cs#L21) but the actual value in JSON is `float`

**Comment by AlyaKoni** [comment]
Looks like a behavioral change in the Json Library:
https://stackoverflow.com/questions/17745866/how-can-i-restore-the-int-deserialization-behavior-after-upgrading-json-net

**Comment by stokuri** [comment]
This change worked for me -> https://github.com/stokuri/pnpcore/commit/0954abc7b405d11598d3cce6032f38f70d60057a

Submitted PR -> https://github.com/pnp/pnpcore/pull/1058
